Automattic/woocommerce-payments
# Handle 402 errors so it won't be hijacked by the hosting provider error page

Fixes #8080, #8079, #6546

#### Changes proposed in this Pull Request

Stripe typically returns a 402 status code when a payment method is invalid. However, this standard response can be intercepted by some hosting companies, leading to a significant issue. These companies replace the expected 402 error with their custom error pages, formatted differently from what the client anticipates.

This PR resolves this problem by returning 400 status code instead of 402, on the AJAX calls where the status code is known that could be hijacked.









#### Testing instructions

Test 1:

1. Create a test site on Jurassic Ninja.
2. Install and activate all the required plugins.
3. Install the WooCommerce Payments.
4. Complete WooCommerce Payments KYC flow.
5. As a merchant, navigate to Pages > Add new.
6. Add the Checkout block to the page.
7. Go to the store page and add a product to your cart.
8. Go to the newly created Checkout page.
9. Checkout with any failure cards (i.e 4000000000009995)
10. Error message should be visible.

Test 2:

You can skip steps from 1-4 if you have already configured JN site with WooPayments.
1. Create a test site on Jurassic Ninja.
2. Install and activate all the required plugins.
3. Install the WooCommerce Payments.
4. Complete WooCommerce Payments KYC flow.
5. Go to the My Account > Add payment method page.
6. Fill the form in with card number 4000000000000002.
7. Click on the "Add payment method" button.
8. Error message should be visible.

-------------------


**Post merge**
